lib/sidekiq/client.rb

```diff
@@ -101,9 +96,12 @@ def push_bulk(items)
       raise ArgumentError, "Job 'at' must be a Numeric or an Array of Numeric timestamps" if at && (Array(at).empty? || !Array(at).all? { |entry| entry.is_a?(Numeric) })
       raise ArgumentError, "Job 'at' Array must have same size as 'args' Array" if at.is_a?(Array) && at.size != args.size
 
+      jid = items.delete("jid")
+      raise ArgumentError, "Explicitly passing 'jid' when pushing more than one job is not supported" if jid && args.size > 1
```

Comment:
This can in theory be a breaking change if some application is using `push_bulk` and explicitly passing `jid`. Before this `jid` would be simply ignored. I'm not sure here. I think that it makes sense to have this explicit check. And explicitly passing `jid` doesn't make much sense, but who knows what happens in the wild.

Reply:
I think it's ok to break undocumented impl details. You are right, passing JID doesn't make a lot of sense in this context.